bentoml/configuration/default_bentoml.yml

```diff
@@ -0,0 +1,42 @@
+#
+# default_bentoml.yml is currently working-in-progress. Configuration changes
+# should still be made to default_bentoml.cfg.
+#
+
+api_server:
+  port: 5000
+  enable_microbatch: False
+  run_with_ngrok: False
+  enable_swagger: True
+  enable_metrics: True
+  enable_feedback: True
+  max_request_size: 20971520
+
+  #
+  # Number of worker threads to be configured for the API server. If Null is
+  # configured, then BentoML will determine an optimal number of worker threads
+  # automatically, see `bentoml.configuration.containers.BentoMLContainer.
+  # api_server_workers` for details.
+  #
+  workers: Null
```

Comment:
we discussed changing this default worker number to 1, do you want to make that change in this PR? I can help update related documentation before the next release and make sure we communicate the breaking change with users in the release note

Reply:
Sure. I can include that change in this PR. What is the reason for changing default workers to 1? The current logic based on CPU cores seems more correct.

Reply:
@ssheng it is because the current calculation based on CPU doesn't really make sense for many use cases and caused lots of confusion for the user. I think worker=1 as a default makes it more obvious that it's a deployment option that users should decide. Plus that 1 is a better default than a variable number associate with CPU: with our adaptive micro-batching layer, `worker=1` still allows the framework itself to take advantage of multiple CPUs when applicable.

Reply:
@parano I would like to learn more about the use cases that don't work with CPU-based worker count. IMHO, users are not best equipped to determine the number of works. The configuration would be different from environment to environment. In an async system like gunicorn, we are in a good position to determine worker count on user's behalf and a constant * the number of cores is typically a very good proxy.